src/Illuminate/Database/Schema/Grammars/ChangeColumn.php

```diff
@@ -119,17 +128,35 @@ protected static function getDoctrineColumn(Table $table, Fluent $fluent)
      */
     protected static function getDoctrineColumnChangeOptions(Fluent $fluent)
     {
-        $options = ['type' => static::getDoctrineColumnType($fluent['type'])];
+        $doctrineType = static::getDoctrineColumnType($fluent['type']);
 
-        if (in_array($fluent['type'], ['text', 'mediumText', 'longText'])) {
+        $options = ['type' => $doctrineType];
+
+        if ($doctrineType->getName() === Types::TEXT) {
             $options['length'] = static::calculateDoctrineTextLength($fluent['type']);
         }
 
+        if ($fluent['type'] === 'binary') {
+            $options['length'] = AbstractMySQLPlatform::LENGTH_LIMIT_BLOB;
+        }
+
         if ($fluent['type'] === 'char') {
             $options['fixed'] = true;
         }
 
-        if (static::doesntNeedCharacterOptions($fluent['type'])) {
+        if ($fluent['type'] === 'timestamp') {
+            $options['platformOptions'] = [
+                'version' => true,
```

Comment:
What does this do?

Reply:
What does "version" = `true` even mean though?

Reply:
They use this option on [`MySQL`](https://github.com/doctrine/dbal/blob/36b9ffc15c31598baa2a37b1abdd5b860d5c164f/src/Platforms/AbstractMySQLPlatform.php#L290-L297) and [`DB2` ](https://github.com/doctrine/dbal/blob/3.4.x/src/Platforms/DB2Platform.php#L293-L300) only on `datetime` type. That's the only way to convert `datetime` to `timestamp`. but I don't know why the call it "version" if that's your question.